### DISCUSSION TITLE
drop declarative-services classpath option

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-pde/eclipse.pde#252] [Ideas]
The declarative services plugin has an option to either automatically include the ds-annotations jar on the classpath or not, and this must be enabled explicitly at the moment. 

With the osgi-bundle/version-annotations added recently we simply add those automatically that has a lot of benefits:

1. Users can control what to include by there target platform, so no need to upgrade / manage this on PDE itself
2. It "just works" so there is no need to specially instruct users how to enable/manage those, check in config files and so on

Because of this, I'd like to suggest we drop this configuration option from the declarative services plugin and instead migrate handling of ds-annotations as with the bundle/version annotations, this will also prevent users from import-package those (either by accident or because PDE suggest this because of missing imports) and will make adoption of annotation based DS components more convenient, user-friendly and visible to the developer.

There is just one thing we "loose" here, that is that PDE currently allows placing an older version of the annotations on the classpath, but I think we will not miss anything here:

1. From a consumer perspective they are backward-compatible 
2. I can't think of a good use-case where one wants an older version of the annotations itself
3. The DS plugin tooling already uses internaly a way to determine if an annotation is placed on valid items regradless of the classpath, so if a user wants he can still code against older definitions
4. PDE actually is already a way behind, DS ist at version 1.5 while PDE only supports 1.3, thus the SCR-Impl is already capable of processing much more than PDE currently supports
5. If a user has really a hard requirement for a DS annotation version it could be controlled by the target itself

**Comment by vogella** [comment]
+1

**Comment by HannesWell** [comment]
Sounds good. +1 from me.
IIRC for the other pde.ds settings the defaults should be fine.
So with the proposed change it should not be necessary any more to add the pde.ds settings per project.

**Reply by laeubi** [reply]
> So with the proposed change it should not be necessary any more to add the pde.ds settings per project.

Because of how the processing works this could slow down eclipse, so I would as a first step still needed to enable the *processing* of the annotations, but they will always be there to be used on the classpath.

**Reply by HannesWell** [reply]
Actually effectivly they are enabeled with the addition of the PDE-DS builder.
But it would be indeed even better if it would not have to be enabeled at all. But yes the annotation processing has to be Speed Up first.
